python/hsfs/core/vector_server.py

```diff
@@ -296,29 +291,62 @@ def deserialize_complex_features(self, feature_schemas, row_dict):
             row_dict[feature_name] = schema.read(decoder)
         return row_dict
 
-    def refresh_mysql_connection(self, external):
+    def refresh_mysql_connection(self):
         try:
-            with self.prepared_statement_engine.connect():
+            with self._prepared_statement_engine.connect():
                 pass
         except exc.OperationalError:
-            self._set_mysql_connection(external)
+            self._set_mysql_connection()
 
     def _make_preview_statement(self, statement, n):
         return text(statement.text[: statement.text.find(" WHERE ")] + f" LIMIT {n}")
 
-    def _set_mysql_connection(self, external):
+    def _set_mysql_connection(self):
         online_conn = self._storage_connector_api.get_online_connector()
-        mysql_engine = util.create_mysql_engine(online_conn, external)
-        self.prepared_statement_engine = mysql_engine
+        self._prepared_statement_engine = util.create_mysql_engine(
+            online_conn, self._external
+        )
 
-    @staticmethod
-    def _apply_transformation(transformation_fns, row_dict):
-        for feature_name in transformation_fns:
+    def _generate_vector(self, result_dict):
+        vector = []
+        for feature in self._features:
+            if feature.label:
+                # Skip the label
+                continue
+
+            if feature.name not in result_dict:
+                raise Exception(
+                    "Feature "
+                    + feature.name
+                    + " is missing from vector. "
+                    + "Please provide the required entry to retrieve it from the feature store "
+                    + " or provide the feature as passed_feature"
+                )
+
+            vector.append(result_dict[feature.name])
+
+        return vector
+
+    def _apply_transformation(self, row_dict):
+        for feature_name in self._transformation_functions:
             if feature_name in row_dict:
-                transformation_fn = transformation_fns[feature_name].transformation_fn
+                transformation_fn = self._transformation_functions[
+                    feature_name
+                ].transformation_fn
                 row_dict[feature_name] = transformation_fn(row_dict[feature_name])
         return row_dict
 
+    @staticmethod
+    def _get_sorted_results(pknames, entries, results):
+        sorted_results = []
+        for e in entries:
+            for row in results:
+                if all([e[pkname] == row[pkname] for pkname in pknames]):
```

Comment:
Somehow this line (344) doesn't work: I get exception `KeyError: 'cc_num'`.  `cc_num` here is PK:
```
feature_view._vector_server.get_feature_vectors(entry = [{"cc_num": 4032019521897961}, {"cc_num": 4007551996724966}])
```